projects/angular-split/src/lib/split/split.component.ts

```diff
@@ -164,22 +165,25 @@ export class SplitComponent {
   constructor() {
     if (isDevMode()) {
       // Logs warnings to console when the provided areas sizes are invalid
-      effect(() => {
-        // Special mode when no size input was declared which is a valid mode
-        if (this.unit() === 'percent' && this._visibleAreas().every((area) => area.size() === 'auto')) {
-          return
-        }
+      afterRenderEffect({
+        read: () => {
+          // Special mode when no size input was declared which is a valid mode
+          if (this.unit() === 'percent' && this._visibleAreas().every((area) => area.size() === 'auto')) {
+            return
+          }
 
-        areAreasValid(this._visibleAreas(), this.unit(), true)
+          areAreasValid(this._visibleAreas(), this.unit(), true)
+        },
       })
     }
 
-    // Responsible for updating grid template style. Must be this way and not based on HostBinding
-    // as change detection for host binding is bound to the parent component and this style
-    // is updated on every mouse move. Doing it this way will prevent change detection cycles in parent.
-    effect(() => {
-      const gridTemplateColumnsStyle = this.gridTemplateColumnsStyle()
-      this.renderer.setStyle(this.elementRef.nativeElement, 'grid-template', gridTemplateColumnsStyle)
+    // we are running this after Angular has completed its CD loop
+    // as we are updating the style of the host, and we don't want to re-trigger the CD loop
+    // doing this in the host of the component would retrigger the CD too many times
+    afterRenderEffect({
+      read: () => {
+        this.elementRef.nativeElement.style.gridTemplate = this.gridTemplateColumnsStyle()
```

Comment:
Isn't `Rendered2` the preferred way when doing direct DOM operations?

Reply:
Not anymore.